convert2rhel.py

```diff
@@ -41,41 +66,83 @@ def create_report(reportfile, log_destination_location):
     where the rhc-worker-script parse the result of the c2r run
     """
     create_log_folder()
-    cmd = ["cp", reportfile, log_destination_location]
-    process = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, bufsize=1)
-    output = ""
-    for line in iter(process.stdout.readline, b""):
-        line = line.decode("utf8")
-        output += line
-
-    # Wait for the process to end
-    process.wait()
-
-    if process.returncode != 0:
-        print(output)
+    try:
+        with open(log_destination_location, "w") as f:
+            f.write(json.dumps(reportfile))
+    except:
         raise RuntimeError("Mock failed to create a report")
 
 
+'''
+    Crafts a report adding all entries from :path_issues to the
+    base reportfile located in :reportfile_path
+
+    :param reportfile_path: Path to the base reportfile
+    :param report_issues: Dict with the paths to the reportfiles to add
+                        Dict has to be str -> str.
+'''
+def craft_report(reportfile_path, report_issues):
+    BASE_REPORT_CRAFT_FOLDER = BASE_REPORT_DATA_FOLDER + "craft/"
+
+    reportfile = {}
+
+    # Get the base reportfile
+    with open(reportfile_path, mode="r") as f:
+        reportfile = json.load(f)
+
+    # Push all report_issues into base reportfile
+    for report_issue in report_issues.values():
+        REPORT_CRAFT_FOLDER = BASE_REPORT_CRAFT_FOLDER + report_issue
+
+        with open(REPORT_CRAFT_FOLDER + ".json", mode="r") as f:
+            json_issue = json.load(f)
+
+        if REPORT_STATUS_ORDER.index(json_issue["status"]) > REPORT_STATUS_ORDER.index(reportfile["status"]):
```

Comment:
Maybe some coment on what or why you doing this would be nice. I suppose you want to put more critical reports on the top, right?

Reply:
This is a bit hacky. The report needs a overall status, and that status is the most critical from all the issues. This only saves the most critical status as the status to report. I added a comment explaining it in my last commit [9870ab6](https://github.com/oamg/c2r-mock-utility/pull/10/commits/9870ab6ce5870083e578a772b9d27fcb451bf6c0).